test/extended/operators/certs.go

```diff
@@ -49,6 +57,123 @@ var _ = g.Describe("[sig-arch][Late]", func() {
 		o.Expect(err).NotTo(o.HaveOccurred())
 	})
 
+	g.It("all tls artifacts must be registered", func() {
+
+		ctx := context.Background()
+		kubeClient := oc.AdminKubeClient()
+		if ok, _ := exutil.IsMicroShiftCluster(kubeClient); ok {
+			g.Skip("microshift does not auto-collect TLS.")
+		}
+
+		nodes := map[string]int{}
+		controlPlaneLabel := labels.SelectorFromSet(map[string]string{"node-role.kubernetes.io/control-plane": ""})
+		nodeList, err := kubeClient.CoreV1().Nodes().List(ctx, metav1.ListOptions{LabelSelector: controlPlaneLabel.String()})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		for i, node := range nodeList.Items {
+			nodes[node.Name] = i
+		}
+
+		actualPKIContent, err := certgraphanalysis.GatherCertsFromPlatformNamespaces(ctx, kubeClient)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		expectedPKIContent, err := certs.GetPKIInfoFromEmbeddedOwnership(ownership.PKIOwnership)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		violationsPKIContent, err := certs.GetPKIInfoFromEmbeddedOwnership(ownership.PKIViolations)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		newTLSRegistry := &certgraphapi.PKIRegistryInfo{}
+
+		for _, currCertKeyPair := range actualPKIContent.InClusterResourceData.CertKeyPairs {
+			currLocation := currCertKeyPair.SecretLocation
+			if _, err := certgraphutils.LocateCertKeyPair(currLocation, violationsPKIContent.CertKeyPairs); err == nil {
+				continue
+			}
+
+			_, err := certgraphutils.LocateCertKeyPair(currLocation, expectedPKIContent.CertKeyPairs)
+			if err != nil {
+				newTLSRegistry.CertKeyPairs = append(newTLSRegistry.CertKeyPairs, currCertKeyPair)
+			}
+		}
+
+		for _, currCABundle := range actualPKIContent.InClusterResourceData.CertificateAuthorityBundles {
+			currLocation := currCABundle.ConfigMapLocation
+			if _, err := certgraphutils.LocateCertificateAuthorityBundle(currLocation, violationsPKIContent.CertificateAuthorityBundles); err == nil {
+				continue
+			}
+
+			_, err := certgraphutils.LocateCertificateAuthorityBundle(currLocation, expectedPKIContent.CertificateAuthorityBundles)
+			if err != nil {
+				newTLSRegistry.CertificateAuthorityBundles = append(newTLSRegistry.CertificateAuthorityBundles, currCABundle)
+			}
+		}
+
+		if len(newTLSRegistry.CertKeyPairs) > 0 || len(newTLSRegistry.CertificateAuthorityBundles) > 0 {
+			registryString, err := json.MarshalIndent(newTLSRegistry, "", "  ")
+			if err != nil {
+				//g.Fail("Failed to marshal registry %#v: %v", newTLSRegistry, err)
+				testresult.Flakef("Failed to marshal registry %#v: %v", newTLSRegistry, err)
+			}
+			// TODO: uncomment when test no longer fails and enhancement is merged
+			//g.Fail(fmt.Sprintf("Unregistered TLS certificates:\n%s", registryString))
+			testresult.Flakef(fmt.Sprintf("Unregistered TLS certificates found:\n%s\nSee tls/ownership/README.md in origin repo", registryString))
+		}
+	})
+
+	g.It("all registered tls artifacts must have expected metadata", func() {
```

Comment:
either separate Its or separate Bys for handling owners separate from descriptions separate from "whatever else".